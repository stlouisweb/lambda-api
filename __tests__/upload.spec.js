import AWS from 'aws-sdk-mock';
import { promisify } from 'util';
import lambda from '../src/upload';
import eventStub from './stubs/eventHttpApiGateway.json';

const handler = promisify(lambda);

describe(`S3 Success mock response`, () => {
  beforeAll(() => {
    AWS.mock('S3', 'getSignedUrl', (method, _, callback) => {
      callback(null, {
        data: 'https://example.com',
      });
    });
  });

  afterEach(() => {
    delete process.env.BUCKET;
    delete process.env.REGION;
  });

  afterAll(() => {
    AWS.restore('S3');
  });

  test(`Require environment variables`, () => {
    const event = {};
    const context = {};

    expect(handler(event, context)).rejects.toThrow(
      `Missing required environment variables: BUCKET, REGION`
    );
  });

  test(`Require a header "x-amz-meta-key"`, () => {
    process.env.BUCKET = 'foo';
    process.env.REGION = 'bar';
    const event = {};
    const context = {};

    const result = handler(event, context);
    result.then(data => expect(data).toMatchSnapshot());
  });

  test(`Replies back with a JSON object for a signed upload on succcess`, () => {
    process.env.BUCKET = 'foo';
    process.env.REGION = 'bar';
    const event = eventStub;
    const context = {};

    const result = handler(event, context);
    result.then(data => expect(data).toMatchSnapshot());
  });
});

describe(`Service aws-node-singned-uploads: S3 mock for failed operations`, () => {
  beforeAll(() => {
    AWS.mock('S3', 'getSignedUrl', (method, _, callback) => {
      callback(`S3 failed`);
    });
  });

  afterEach(() => {
    delete process.env.BUCKET;
    delete process.env.REGION;
  });

  afterAll(() => {
    AWS.restore('S3');
  });

  test(`Correctly handles error messages from S3`, () => {
    process.env.BUCKET = 'foo';
    process.env.REGION = 'bar';
    const event = eventStub;
    const context = {};

    const result = handler(event, context);
    result.then(data => expect(data).toMatchSnapshot());
  });
});
