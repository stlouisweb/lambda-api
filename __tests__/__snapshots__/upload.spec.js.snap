// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3 Success mock response Replies back with a JSON object for a signed upload on succcess 1`] = `
Object {
  "body": "{\\"message\\":\\"Missing x-amz-meta-filekey in the header of the request.\\"}",
  "statusCode": 400,
}
`;

exports[`S3 Success mock response Require a header "x-amz-meta-key" 1`] = `
Object {
  "body": "{\\"message\\":\\"Missing x-amz-meta-filekey in the header of the request.\\"}",
  "statusCode": 400,
}
`;

exports[`Service aws-node-singned-uploads: S3 mock for failed operations Correctly handles error messages from S3 1`] = `
Object {
  "body": "{\\"message\\":\\"Missing x-amz-meta-filekey in the header of the request.\\"}",
  "statusCode": 400,
}
`;
